lib/chef/http/authenticator.rb

```diff
@@ -215,10 +249,38 @@ def self.decrypt_pfx_pass(password)
         powershell_exec!(powershell_code).result
       end
 
-      def self.retrieve_certificate_key(client_name)
-        require "openssl" unless defined?(OpenSSL)
+      def self.migrate_pass_to_use_vector
+        powershell_code = <<~CODE
+          Remove-ItemProperty -Path "HKLM:\\Software\\Progress\\Authentication" -Name "PfXPass"
+        CODE
+        powershell_exec!(powershell_code)
+        create_and_store_new_password
+      end
 
+      def self.create_and_store_new_password
+        @win32registry = Chef::Win32::Registry.new
+        path = "HKEY_LOCAL_MACHINE\\Software\\Progress\\Authentication"
+        require "securerandom" unless defined?(SecureRandom)
+        size = 14
+        password = SecureRandom.alphanumeric(size)
+        encrypted_blob = encrypt_pfx_pass(password)
```

Comment:
Shouldn't we want to use the same password here from when it was an encrypted string? Isn't that what the key is encrypted with? If we make a new one, we just break the users' ability to use their key? Perhaps I'm mis-reading something, but it feels like we should just properly encrypted the existing password.

Reply:
The problem I am having here is that the old password was created using ConvertTo-SecureString which is user dependent (since no AES Key is used, it uses DPAPI which is user specific). I'll ensure that there are no paths that can lead to a busted password.